### `Problem 1`: Write a program that will give you in hand monthly salary after deduction on CTC - HRA(10%), DA(5%), PF(3%) and taxes deduction as below:

> Salary(Lakhs) : Tax(%)

*   Below 5 : 0%
*   5-10 : 10%
*   10-20 : 20%
*   aboove 20 : 30%

In [4]:
# Write code here
annual_ctc = float(input("Enter your annual salary in lakhs: ")) * 100000

monthly_ctc = annual_ctc/12

deduction = monthly_ctc * (0.10 + 0.05 + 0.03)

if annual_ctc < 500000:
    annual_tax_rate = 0
elif annual_ctc <= 1000000:
    annual_tax_rate = 0.10
elif annual_ctc <= 2000000:
    annual_tax_rate = 0.20
else:
    annual_tax_rate = 0.30
    
monthly_tax = (annual_ctc * annual_tax_rate) / 12

# In-hand Salary
in_hand = monthly_ctc - deduction - monthly_tax

print(f"Your Monthly In-hand Salary is: {in_hand:.2f}")


Your Monthly In-hand Salary is: 30000.00


### `Problem 2`: Write a program that take a user input of three angles and will find out whether it can form a triangle or not.

In [6]:
# Write code here
# Taking input of three angles from the user
angle1 = float(input("Enter first angle: "))
angle2 = float(input("Enter second angle: "))
angle3 = float(input("Enter third angle: "))

# Checking if the sum of angles is 180 and each angle is greater than 0
# A triangle can only be formed if the sum of angles is exactly 180 degrees 
# and all angles have a positive value.
if (angle1 + angle2 + angle3 == 180) and (angle1 > 0 and angle2 > 0 and angle3 > 0):
    print("Yes, these angles can form a triangle!")
else:
    print("No, these angles cannot form a triangle.")

No, these angles cannot form a triangle.


### `Problem 3`: Write a program that will take user input of cost price and selling price and determines whether its a loss or a profit.

In [10]:
# Write code here
# Taking input from the user for Cost Price and Selling Price
cost_price = float(input("Enter the cost price: "))
selling_price = float(input("Enter the selling price: "))

# Comparing the prices to determine profit or loss
if selling_price > cost_price:
    profit = selling_price - cost_price
    print(f"It is a profit of {profit:.2f}")

elif cost_price > selling_price:
    loss = cost_price - selling_price
    print(f"It is a loss of {loss:.2f}")

else:
    print("There is no profit and no loss.")

It is a loss of 370.00


### `Problem 4`: Write a menu-driven program -
1. cm to ft
2. km to miles
3. USD to PKR
4. exit

In [24]:
# Write code here
while True:
    # Display the menu options
    print("\n--- Unit Converter Menu ---")
    print("1. Centimeters to Feet")
    print("2. Kilometers to Miles")
    print("3. USD to PKR")
    print("4. Exit")
    
    choice = input("Enter your choice (1/2/3/4): ")
    
    # Handle the user's choice
    if choice == '1':
        cm = float(input("Enter length in cm: "))
        ft = cm * 0.0328084
        print(f"{cm} cm is equal to {ft:.2f} feet.")
        
    elif choice == '2':
        km = float(input("Enter distance in km: "))
        miles = km * 0.621371
        print(f"{km} km is equal to {miles:.2f} miles.")
        
    elif choice == '3':
        usd = float(input("Enter amount in USD: "))
        # Assuming 1 USD = 278 PKR as an example exchange rate
        pkr = usd * 278.0
        print(f"${usd} is equal to {pkr:.2f} PKR.")
        
    elif choice == '4':
        print("Exiting the program. Goodbye!")
        break # Exit the loop
        
    else:
        print("Invalid choice, please try again.")


--- Unit Converter Menu ---
1. Centimeters to Feet
2. Kilometers to Miles
3. USD to PKR
4. Exit
40.0 cm is equal to 1.31 feet.

--- Unit Converter Menu ---
1. Centimeters to Feet
2. Kilometers to Miles
3. USD to PKR
4. Exit
$500.0 is equal to 139000.00 PKR.

--- Unit Converter Menu ---
1. Centimeters to Feet
2. Kilometers to Miles
3. USD to PKR
4. Exit
Exiting the program. Goodbye!


### `Problem 5` - Exercise 12: Display Fibonacci series up to 10 terms.

*Note: The Fibonacci Sequence is a series of numbers. The next number is found by adding up the two numbers before it. The first two numbers are 0 and 1. For example, 0, 1, 1, 2, 3, 5, 8, 13, 21. The next number in this series above is 13+21 = 34*

In [1]:
# Write code here

# Number of terms to be printed in the Fibonacci series
n = 10

# Initialize the first two terms of the series: 0 and 1
a, b = 0, 1 
print(a, b, end=" ")

# Start the loop from 2 up to n, as the first two terms are already printed
for i in range(2, n):
    # Calculate the next term by adding the previous two
    temp = a + b
    print(temp, end=" ")
    
    # Update 'a' and 'b' for the next iteration:
    # 'a' takes the value of 'b'
    # 'b' takes the value of the newly calculated term (temp)
    a = b
    b = temp


0 1 1 2 3 5 8 13 21 34 

### `Problem 6` - Find the factorial of a given number.

Write a program to use the loop to find the factorial of a given number.

The factorial (symbol: `!`) means to multiply all whole numbers from the chosen number down to 1.

For example: calculate the factorial of 5

```bash
5! = 5 × 4 × 3 × 2 × 1 = 120
```

Output:

```bash
120
```

In [ ]:
# Write code here
n = int(input("Enter number: "))

fact = 1    # if we insialize it with 0 then in all cases answer will be zero thats why 1
for i in range(1,n+1):
    fact *= i   # multpling each number
    
print(fact)

120


### `Problem 7` - Reverse a given integer number.

Example:

`Input:`

```bash
76542
```

`Output:`

```bash
24567
```

In [45]:
# Write code here

# First Method
# Taking integer input from the user
number = int(input("Enter an integer: "))

# Initialize the reversed number
reversed_number = 0

# Use a loop to reverse the digits
temp = number
while temp > 0:
    # Get the last digit
    digit = temp % 10
    
    # Build the reversed number
    reversed_number = (reversed_number * 10) + digit
    
    # Remove the last digit from the number
    temp = temp // 10

print(f"The reversed number is: {reversed_number}")

# Second Method
# convert int --> str then use s1[::-1] then comback to interger

str1 = str(number)
reversed = int(str1[::-1])

print(reversed)


The reversed number is: 54321
54321


### `Problem 8`: Take a user input as integer N. Find out the sum from 1 to N. If any number if divisible by 5, then skip that number. And if the sum is greater than 300, don't need to calculate the sum further more. Print the final result. And don't use for loop to solve this problem.

**Example 1:**

`Input:`

```bash
30
```

`Output:`

```bash
276
```

In [61]:
# Write code here
n = int(input("Enter a number: "))

total_sum = 0
current_number = 1

while current_number <= n:
    # Check if number is not divisible by 5
    if current_number % 5 != 0:
        # Check if adding the next number will exceed 300
        if total_sum + current_number > 300:
            break # Stop before adding the number
        total_sum += current_number
        
    current_number += 1

print(f"The final result is: {total_sum}")

The final result is: 276


### `Problem 9`: Write a program that keeps on accepting a number from the user until the user enters Zero. Display the sum and average of all the numbers.

In [67]:
# Write code here

# Initialize variables
total_sum = 0
count = 0

print("Enter numbers to calculate sum and average (Enter 0 to stop):")

while True:
    # Accept input from the user
    number = float(input("Enter a number: "))
    
    # Check if the input is zero to break the loop
    if number == 0:
        break
    
    # Add number to sum and increment the count
    total_sum += number
    count += 1

# Calculate and display the results
if count > 0:
    average = total_sum / count
    print(f"\nTotal Sum: {total_sum}")
    print(f"Average: {average:.2f}")
else:
    
    print("No numbers were entered.")

Enter numbers to calculate sum and average (Enter 0 to stop):

Total Sum: 9.0
Average: 3.00


### `Problem 9`: Write a program which will find all such numbers which are divisible by 7 but are not a multiple of 5, between 2000 and 3200 (both included). The numbers obtained should be printed in a comma-separated sequence on a single line.

In [73]:
# Write code here

for i in range(2000, 3201):
    # Check if number is divisible by 7 and not by 5
    if i % 7 == 0 and i % 5 != 0:
        print(i,end = ",")



2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199,

### `Problem 10`: Write a program, which will find all such numbers between 1000 and 3000 (both included) such that each digit of the number is an even number. The numbers obtained should be printed in a space-separated sequence on a single line.



In [77]:
# Write code here

for i in range(1000,3001):
    s = str(i)
    
    
    if (int(s[0]) % 2 == 0) and (int(s[1]) % 2 == 0) and (int(s[2]) % 2 == 0) and (int(s[3]) % 2 == 0):
        # If all digits are even, print the number
        print(i, end=" ")
    

2000 2002 2004 2006 2008 2020 2022 2024 2026 2028 2040 2042 2044 2046 2048 2060 2062 2064 2066 2068 2080 2082 2084 2086 2088 2200 2202 2204 2206 2208 2220 2222 2224 2226 2228 2240 2242 2244 2246 2248 2260 2262 2264 2266 2268 2280 2282 2284 2286 2288 2400 2402 2404 2406 2408 2420 2422 2424 2426 2428 2440 2442 2444 2446 2448 2460 2462 2464 2466 2468 2480 2482 2484 2486 2488 2600 2602 2604 2606 2608 2620 2622 2624 2626 2628 2640 2642 2644 2646 2648 2660 2662 2664 2666 2668 2680 2682 2684 2686 2688 2800 2802 2804 2806 2808 2820 2822 2824 2826 2828 2840 2842 2844 2846 2848 2860 2862 2864 2866 2868 2880 2882 2884 2886 2888 

### `Problem 11`: A robot moves in a plane starting from the original point (0,0). The robot can move toward UP, DOWN, LEFT and RIGHT with a given steps.
The trace of robot movement is shown as the following:
```
UP 5
DOWN 3
LEFT 3
RIGHT 2
!
```
> The numbers after the direction are steps.

> `!` means robot stop there.

**Please write a program to compute the distance from current position after a sequence of movement and original point.**

*If the distance is a float, then just print the nearest integer.*

Example:

`Input`:
```
UP 5
DOWN 3
LEFT 3
RIGHT 2
!
```
`Output`:
```
2
```

In [7]:
# Write code here


x ,y = 0,0
print("Enter commond (e.g UP 5) or '!' for exit")
while True:
    command = input()
    
    if command == "!":
        break
    
    parts = command.split()
    direction = parts[0]
    steps = int(parts[1])
    
    if direction.upper() == "UP":
        y += steps
    
    elif direction.upper() == "DOWN":
        y -= steps
        
    elif direction.upper() == "LEFT":
        x -= steps
    
    elif direction.upper() == "RIGHT":
        x += steps
    else:
        print("Invalid input.")
    
distance = (x**2 + y**2)**0.5

print(round(distance))


Enter commond (e.g UP 5) or '!' for exit
2


### `Problem 12`:Write a program to print whether a given number is a prime number or not

In [96]:
# Write code here
n = int(input("Enter a number: "))

# Numbers less than 2 are not prime
if n < 2:
    print("Not a prime number.")
else:
    is_prime = True
    # Check for divisibility from 2 up to n-1
    for i in range(2, n):
        if n % i == 0:
            is_prime = False
            break # Once we find a divisor, we can stop the loop
    
    if is_prime:
        print("Yes, it is a prime number.")
    else:
        print("Not a prime number.")   

Yes, it is a prime number.


### `Problem 13`:Print all the Armstrong numbers in a given range.
Range will be provided by the user<br>
Armstrong number is a number that is equal to the sum of cubes of its digits. For example 0, 1, 153, 370, 371 and 407 are the Armstrong numbers.

In [9]:
# Write code here

# Take the starting and ending range as input from the user[cite: 1]
start = int(input("Enter the starting number of the range: "))
end = int(input("Enter the ending number of the range: "))

print(f"\nArmstrong numbers between {start} and {end} are:")

# Loop through each number in the given range
for num in range(start, end + 1):
    # Convert number to string to find the total number of digits (power)
    num_str = str(num)
    power = len(num_str)
    
    sum_of_powers = 0
    # Calculate the sum of each digit raised to the power of the total number of digits
    for digit in num_str:
        sum_of_powers += int(digit) ** power
        
    # If the sum equals the original number, it is an Armstrong number[cite: 1]
    if sum_of_powers == num:
        print(num, end=" ")
    


Armstrong numbers between 10 and 1000 are:
153 370 371 407 

### `Problem 14`:Calculate the angle between the hour hand and minute hand.

Note: There can be two angles between hands; we need to print a minimum of two. Also, we need to print the floor of the final result angle. For example, if the final angle is 10.61, we need to print 10.

Input:<br>
H = 9 , M = 0<br>
Output:<br>
90<br>
Explanation:<br>
The minimum angle between hour and minute
hand when the time is 9 is 90 degress.

In [ ]:
# Write code here

### `Problem 15`:Given two rectangles, find if the given two rectangles overlap or not. A rectangle is denoted by providing the x and y coordinates of two points: the left top corner and the right bottom corner of the rectangle. Two rectangles sharing a side are considered overlapping. (L1 and R1 are the extreme points of the first rectangle and L2 and R2 are the extreme points of the second rectangle).

Note: It may be assumed that the rectangles are parallel to the coordinate axis.

<img src='https://www.geeksforgeeks.org/wp-content/uploads/rectanglesOverlap.png' width='300' height='200'>

In [ ]:
# Write code here